src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/AddVenueController.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Flurl;
+using FormFlow;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue
+{
+    [Route("venues/add")]
+    [JourneyMetadata(
+        journeyName: "AddVenue",
+        stateType: typeof(AddVenueJourneyModel),
+        appendUniqueKey: true,
+        requestDataKeys: "providerId?")]
+    [RequireProviderContext]
+    public class AddVenueController : Controller
+    {
+        private const string ReturnUrlJourneyInstancePropertyKey = "returnUrl";
+
+        private readonly IMediator _mediator;
+        private readonly JourneyInstanceProvider _journeyInstanceProvider;
+        private readonly IProviderContextProvider _providerContextProvider;
+        private JourneyInstance<AddVenueJourneyModel> _journeyInstance;
+
+        public AddVenueController(
+            IMediator mediator,
+            JourneyInstanceProvider journeyInstanceProvider,
+            IProviderContextProvider providerContextProvider)
+        {
+            _mediator = mediator;
+            _journeyInstanceProvider = journeyInstanceProvider;
+            _providerContextProvider = providerContextProvider;
+            _journeyInstance = _journeyInstanceProvider.GetInstance<AddVenueJourneyModel>();
+        }
+
+        [HttpGet("")]
+        public IActionResult Index([FromQuery] string postcode, [FromQuery] string returnUrl)
+        {
+            _journeyInstance = _journeyInstanceProvider.GetOrCreateInstance(
+                () => new AddVenueJourneyModel(),
+                new PropertiesBuilder()
+                    .Add(ReturnUrlJourneyInstancePropertyKey, returnUrl)
+                    .Build());
+
+            if (!_journeyInstanceProvider.IsCurrentInstance(_journeyInstance))
+            {
+                return RedirectToAction()
+                    .WithProviderContext(_providerContextProvider.GetProviderContext())
+                    .WithJourneyInstanceUniqueKey(_journeyInstance);
+            }
+
+            return View(
+                nameof(PostcodeSearch),
+                new PostcodeSearch.SearchCommand()
```

Comment:
Why are you passing a "Command" to a view? This doesn't make sense to me in the context of the CQRS pattern that's implemented into the existing system.

Reply:
What else would we name it?

Reply:
Surely it would be a ViewModel that inherits from the Command based on previous implementations of this pattern?

Reply:
That pattern is used where the view model needs more data than is in the `Command`.

Reply:
I suppose this is a drawback of the pattern we're using, my only suggestion would be to create a vm wrapper for readability and consistency with the other implementations of this pattern.

Reply:
We don't always have a ViewModel - the Edit journey follows a similar pattern to here where there's just a `Query` and `Command`.